rlog/2023-09-27-gossipimprovements.mdx

```diff
@@ -0,0 +1,214 @@
+---
+title: 'GossipSub Improvements: Evolution of Overlay Design and Message Dissemination in Unstructured P2P Networks'
+date: 2023-09-25 12:00:00
+authors: Umar Farooq, Daniel Kaiser, Tanguy Cizain
+published: true
+slug: GossipSub Improvements
+categories: research
+
+
+toc_min_heading_level: 2
+toc_max_heading_level: 5
+---
+
+GossipSub Improvements: Evolution of Overlay Design and Message Dissemination in Unstructured P2P Networks 
+
+<!--truncate-->
+
+## Introduction
+
+Nodes in an unstructured p2p network form self-organizing overlay(s) on top of the IP infrastructure to facilitate different services like information dissemination,
+query propagation, file sharing, etc. The overlay(s) can be as optimal as a tree-like structure or as enforcing as a fully connected mesh. 
+
+Due to peer autonomy and a trustless computing environment, some peers may deviate from the expected operation or even leave the network.
+At the same time, the underlying IP layer is unreliable. 
+
+Therefore, tree-like overlays are not best suited for reliable information propagation.
+Moreover, tree-based solutions usually result in significantly higher message dissemination latency due to suboptimal branches. 
+
+Flooding-based solutions, on the other hand, result in maximum resilience against adversaries and achieve minimal message dissemination latency because the message propagates through all (including the optimal) paths. 
+Redundant transmissions help maintain the integrity and security of the network in the presence of adversaries and high node failure but significantly increase network-wide bandwidth utilization, cramming the bottleneck links. 
+
+An efficient alternative is to lower the number of redundant transmissions by D-regular broadcasting, where a peer will likely receive (or relay) a message from up to $D$ random peers.
+This mechanism is also termed gossiping or rumor-mongering. 
+
+Publishing through a D-regular overlay triggers approximately $N \times D$ transmissions.
+Reducing $D$ reduces the redundant transmissions but compromises reachability and latency. 
+
+Sharing metadata through a K-regular overlay (where $K > D$) allows nodes to pull missing messages. 

```

Comment:
AFAIK, the K overlay is gossiping (since it happens at fixed intervals), the D overlay is flooding (or maybe another name, since we don't flood to everyone, not sure)

Reply:
0d3dacb71ff98c6f7710c491c7312009fe76f37a